lib/phoenix_live_view/diff.ex

```diff
@@ -465,9 +465,13 @@ defmodule Phoenix.LiveView.Diff do
           traverse(socket, rendered, socket.fingerprints, pending_components, components)
 
         socket = Utils.clear_changed(%{socket | fingerprints: component_prints})
-        {socket, pending_components, Map.put(component_diffs, cid, diff), components}
+        if diff == %{} do
+          {socket, pending_components, component_diffs, components}
+        else
+          {socket, pending_components, Map.put(component_diffs, cid, diff), components}
+        end
       else
-        {socket, pending_components, Map.put(component_diffs, cid, %{}), components}
+        {socket, pending_components, component_diffs, components}
```

Comment:
To clarify, this change would be safe to apply. The one directly above isn't and the other one I am not so sure.